src/pyxplor/plot_time_series.py

```diff
@@ -8,54 +10,71 @@ def plot_time_series(input_df: pd.DataFrame,
-    each aggregated at a given frequency. It creates a single figure with subplots for each variable, 
-    allowing for easy comparison of trends, seasonality, and other time-series characteristics. 
-    The function also offers the option to save and display the resulting figure.
+    This function creates a figure with subplots for each specified time series variable,
+    aggregated at a given frequency. It allows for comparisons of trends and seasonality.
 
     Parameters
     ----------
     input_df : pd.DataFrame
-        The DataFrame containing the time-series data. Expected to have at least two columns: 
-        one for dates and others for numeric values.
+        DataFrame containing the time-series data with date and numeric columns.
         
     date_column : str
-        The name of the column in `input_df` that contains date/time information. 
-        This column should be convertible to pandas datetime format.
+        Name of the column containing date/time information in datetime format.
 
     value_columns : list
-        A list of names of the columns in `input_df` that contain the numeric values to be analyzed.
+        List of column names containing numeric values to plot as time series.
 
-    freq : {"D", "W", "M"}, optional
-        The frequency of the time-series data ('D' for daily, 'W' for weekly, etc.). Default is 'D'.
+    freq : str, optional
+        Frequency of data aggregation ('D' for daily, 'W' for weekly, etc.). Default is 'D'.
 
     figsize : tuple[int, int], optional
-        The size of the plot in the format (width, height). Default is (10, 6).
+        Size of the plot as (width, height). Default is (10, 6).
 
     output_path : str, optional
-        Path to save the plot. Defaults to the current working directory.
+        File path to save the plot. If None, the plot is not saved.
 
     super_title : str, optional
-        Super title for the entire plot. Default is "Time Series Analysis".
+        Title for the entire plot. Default is "Time Series Analysis".
 
     super_title_font : int, optional
         Font size for the super title. Default is 14.
 
     Returns
     -------
     None
-
-    Examples
-    --------
-    dates = pd.date_range(start='2020-01-01', end='2020-12-31', freq='D')
-    data = pd.DataFrame({
-        'date': dates,
-        'sales': np.random.rand(len(dates)) * 100,
-        'expenses': np.random.rand(len(dates)) * 50,
-    })
-    plot_time_series(data, 'date', ['sales', 'expenses'], freq='M', figsize=(12, 6))
     """
 
-    pass  # Implementation to be added
+    if not isinstance(input_df, pd.DataFrame):
+        raise ValueError("Input data must be a pandas DataFrame.")
+
+    if date_column not in input_df.columns:
+        raise ValueError(f"Date column '{date_column}' not found in DataFrame.")
+
+    input_df[date_column] = pd.to_datetime(input_df[date_column])
```

Comment:
What if date_column is something that cannot be transformed into a datetime object?

Reply:
I can add more tests